### PR TITLE
fix: loading wallet could crash

### PR DESCRIPTION
**Pull request by obiyankenobi**

### CODE REVIEW

File: src/wallet.js [hunk 1]
```diff
@@ -1571,7 +1573,10 @@ const wallet = {
    */
   txExists(txData) {
     const data = this.getWalletData();
-    return txData.tx_id in data['historyTransactions'];
+    if (data && data['historyTransactions']) {
```
Comment (msbrogli) [inline src/wallet.js]: When can `data` or `data.historyTransactions` be undefined? Maybe we should add a test that fails without this fix.

Comment (obiyankenobi) [reply src/wallet.js]: It happens before we load the wallet.

Added the test.

File: src/api/wallet.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ const walletApi = {
    */
   getAddressHistory(addresses, resolve) {
     const data = {addresses};
-    return createRequestInstance(resolve).get('thin_wallet/address_history', {'params': data}).then((res) => {
+    return createRequestInstance(resolve, 40000).get('thin_wallet/address_history', {'params': data}).then((res) => {
```
Comment (msbrogli) [inline src/api/wallet.js]: I don't think we should increase the time limit.

Comment (obiyankenobi) [reply src/api/wallet.js]: Unless we get the pagination PR (#47) as well, we need to increase it.

Comment (msbrogli) [reply src/api/wallet.js]: Ok. We can change it, but we need to undo it after the PR #47 is merged. So, I left a comment in PR #47 to reduce this time limit.

**Pull request merged**